packages/gateway/src/features/notification/service.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+import { IAuthHeader } from '@opencrvs/commons'
+import { USER_MANAGEMENT_URL, NOTIFICATION_URL } from '@gateway/constants'
+import fetch from '@gateway/fetch'
+import { IUserModelData } from '@gateway/features/user/type-resolvers'
+import { internal } from '@hapi/boom'
+
+const DEFAULT_PAGE_SIZE = 500
+
+async function fetchAllUsersInPage(
+  pageSize: number,
+  pageNumber: number,
+  authHeader: IAuthHeader
+) {
+  const res = await fetch(`${USER_MANAGEMENT_URL}searchUsers`, {
+    method: 'POST',
+    body: JSON.stringify({
+      count: pageSize,
+      skip: pageNumber * pageSize,
+      sortOrder: 'desc'
+    }),
+    headers: {
+      'Content-Type': 'application/json',
+      ...authHeader
+    }
+  })
+
+  if (res.status === 200) {
+    return (await res.json()) as Promise<{
+      total: number
+      results: IUserModelData[]
+    }>
+  } else {
+    throw internal('Something went wrong in user-mgnt server')
```

Comment:
Forward the received error. Otherwise this will become quite difficult to debug in deployed environments